### ISSUE TITLE
Better support for zip with toTable()

### ISSUE THREAD

**Issue by oderwat** [Araq/Nim#2126]
I stumbled over a Rosetta Code example I did not understand at first. Here my thoughts and findings embedded in the code example:

```Nimrod
import tables, sequtils, math
 
let keys = @['a','b','c']
let values = @[1, 2, 3]

# Rosetta Code example by def (modified to use the [T]
proc asKeyVal[T](x: T): auto = cast[seq[tuple[key: char, val: int]]](x)

echo toTable zip(keys, values).asKeyVal

# What I think would maybe a better example for Rosetta 
let table = newTable[char, int](keys.high)
for t in zip(keys, values):
    table[t.a] = t.b

echo table


# A new variant of toTable() which takes two arrays
proc toTable[A, B](keys: seq[A], values: seq[B]): auto =
    result = newTable[A, B]()
    for t in zip(keys, values):
        result[t.a] = t.b

echo toTable(keys, values)

# Then I duplicated the toTable but for key/values names as produced by zip
# I think that should go standard and Rosetta code and practical use is enhanced
proc toTable[A, B](pairs: openArray[tuple[a: A, b: B]]): Table[A, B] =
    ## creates a new hash table that contains the given zip style `pairs`.
    result = initTable[A, B](nextPowerOfTwo(pairs.len+10))
    for a, b in items(pairs): result[a] = b

echo toTable zip(keys, values)
```

EDIT: I updated it with the correct code for the two arrays

**Comment by Araq** [comment]
We need a better way to do ``type((1, 2))`` for tuples with no tuple names, then we can update the stdlib to use these for better interop.

**Comment by Araq** [comment]
Now that we have (int, int) type tuple syntax, can you work on the stdlib to make this happen please?

**Comment by oderwat** [comment]
Hmm... Not sure what to do

Does that mean that in `tables.nim` every result should change from:
```
iterator pairs*[A, B](t: Table[A, B]): tuple[key: A, val: B] = ...
```
to
```
iterator pairs*[A, B](t: Table[A, B]): (A, B) = ...
```

or just arguments:
```
proc toTable*[A, B](pairs: openArray[tuple[key: A, val: B]]): Table[A, B] = ...
```
to
```
proc toTable*[A, B](pairs: openArray[(A, B)]): Table[A, B] = ...
```

Or even all (internal) types?
```
type KeyValuePair[A, B] = tuple[hcode: THash, key: A, val: B]`
```
to
```
type KeyValuePair[A, B] = (THash, A , B) # and accessed with [0], [1], [2] 
```

**Comment by Araq** [comment]
> Does that mean that in tables.nim every result should change from:
> iterator pairs*[A, B](t: Table[A, B]): tuple[key: A, val: B] = ...
> to
> iterator pairs*[A, B](t: Table[A, B]): (A, B) = ...

Yes. And yes, it's a breaking change. Not sure how to handle that transition yet.

**Comment by oderwat** [comment]
So you want to change "type KeyValuePair[A, B] = tuple[hcode: THash, key: A, val: B]`" into the "[0]" indexed version too? This would be a lot less readable though. And I think that most code will not break. Everything I tried (ok maybe not enough) works with those changes. Even all tests in tcollection still work with that PR applied. I would not change the deprecated types (I stopped after some) and leave the named tuples for the implementation part. Just change the interface to the proc and the return values to be anonymous.
